gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenFragment.java

```diff
@@ -316,30 +311,25 @@ private void showOfflineAreas() {
   }
 
   private void onApplyWindowInsets(WindowInsetsCompat insets) {
+    this.insets = insets;
     statusBarScrim.setPadding(0, insets.getSystemWindowInsetTop(), 0, 0);
     toolbarWrapper.setPadding(0, insets.getSystemWindowInsetTop(), 0, 0);
     bottomSheetBottomInsetScrim.setMinimumHeight(insets.getSystemWindowInsetBottom());
     updateNavViewInsets(insets);
-    updateBottomSheetPeekHeight(insets);
   }
 
   private void updateNavViewInsets(WindowInsetsCompat insets) {
     View headerView = navView.getHeaderView(0);
     headerView.setPadding(0, insets.getSystemWindowInsetTop(), 0, 0);
   }
 
-  private void updateBottomSheetPeekHeight(WindowInsetsCompat insets) {
-    double width =
-        getScreenWidth(getActivity())
-            + insets.getSystemWindowInsetLeft()
-            + insets.getSystemWindowInsetRight();
-    double height =
-        getScreenHeight(getActivity())
-            + insets.getSystemWindowInsetTop()
-            + insets.getSystemWindowInsetBottom();
-    double mapHeight = width / COLLAPSED_MAP_ASPECT_RATIO;
-    double peekHeight = height - mapHeight;
-    bottomSheetBehavior.setPeekHeight((int) peekHeight);
+  private int getBottomSheetPeekHeight(WindowInsetsCompat insets) {
+    double screenWidth = getScreenWidth(getActivity());
+    double screenHeight = getScreenHeight(getActivity());
+    double collapseMapHeight = screenWidth / COLLAPSED_MAP_ASPECT_RATIO;
+    int insetTop = insets.getSystemWindowInsetTop();
+    int headerHeight = getView().findViewById(R.id.feature_sheet_header).getHeight();
+    return (int) (screenHeight - insetTop - collapseMapHeight - headerHeight);
```

Comment:
@dturner The resulting map thumbnail aspect ratio appears to be wrong but not really sure why; I'd expect it to be 4:3 as defined by `COLLAPSED_MAP_ASPECT_RATIO` but it appears closer to 1:1. I'm probably doing the math wrong here, but not sure where.

![Screenshot_1587604134](https://user-images.githubusercontent.com/228050/80048195-7c85be00-84dd-11ea-9386-63ad44f206c6.png)

Reply:
If by "map thumbnail" you mean the map area above the bottom sheet it could be the line `screenHeight - insetTop - collapseMapHeight - headerHeight` because presumably what you actually want is `screenHeight  - (insetTop + collapseMapHeight + headerHeight)`. Difficult to debug without step debugging.